master/buildbot/www/authz/roles.py

```diff
@@ -61,6 +61,26 @@ def getRolesFromUser(self, userDetails):
         return []
 
 
+class RolesFromDomain(RolesFromEmails):
+
+    def __init__(self, **kwargs):
+        super(RolesFromEmails, self).__init__()
+
+        self.domain_roles = {}
+        for role, domains in iteritems(kwargs):
+            for domain in domains:
+                self.domain_roles.setdefault(domain, []).append(role)
+
+    def getRolesFromUser(self, userDetails):
+        if 'email' in userDetails:
+            email = userDetails['email']
+            edomain = email.split('@')[-1]
+            if edomain in self.domain_roles:
```

Comment:
you can simplify the whole if block:

    return self.domain_roles.get(edomain, [])